pkg/log/io/out.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+// OutputController manages the output for the CLI
+type OutputController struct {
+	out io.Writer
+
+	formatter formatter
+	decorator decorator
+
+	spinner OktetoSpinner
+}
+
+// newOutputController returns a new logger that writes to stdout
+func newOutputController(out io.Writer) *OutputController {
+	return &OutputController{
+		out:       out,
+		formatter: newTTYFormatter(),
+		decorator: newTTYDecorator(),
+	}
+}
+
+// SetOutputFormat sets the output format
+func (oc *OutputController) SetOutputFormat(output string) {
+	switch output {
+	case "tty":
+		oc.formatter = newTTYFormatter()
```

Comment:
having the same life cycle, does it make sense to unify decorator and formater?

Reply:
formatter needs to be also used into the logger in order to use the `json`. So we'll still need to have it separate for that. If we include the decorator into the formater which method should be the default one? `none`, `default`.

Most of the functions doesn't need decorations, only the ones that needs colors, so calling decorator on a function to add format doesn't make total sense for me. But if you agree I can merge them